src/xrpld/app/tx/detail/InvariantCheck.cpp

```diff
@@ -2297,7 +2297,7 @@ ValidVault::visitEntry(
     }
 
     uint256 const key = (before ? before->key() : after->key());
-    if (sign && balance != zero)
+    if (sign != 0)
```

Comment:
The description of `balance` makes clear that it's a delta. But looking at a line like this in isolation, it's easy to forget. I think it would make sense to rename it to indicate more accurately what it is. Something like `delta`, `balanceDelta`, `balanceChange`, `balanceDifference`....